src/_pytest/mark/legacy.py

```diff
@@ -25,16 +32,16 @@ def __getitem__(self, name):
         return name in self.own_mark_names
 
 
+@attr.s
 class KeywordMapping:
     """Provides a local mapping for keywords.
     Given a list of names, map any substring of one of these names to True.
     """
 
-    def __init__(self, names):
-        self._names = names
+    _names = attr.ib(type=Set[str])
 
     @classmethod
-    def from_item(cls, item):
+    def from_item(cls, item: Union["Node", "Function"]) -> "KeywordMapping":
```

Comment:
`Function` is a subtype of `Node`, so a union of them is redundant.

BTW, is this really `Node` (effectively `Union[Item, Collector]`) or just `Item`?

Reply:
This is because when using only `Node`, `mypy` complained that `Node` has no `function` attribute, that's why I have changed to use a `getattr` instead of `hasattr` in line 58 below.

(Got that advice from here: https://github.com/python/mypy/issues/1424)

Of course I'm open to suggestions! 😁

Reply:
I think the `getattr` is the fix, and the `Union` isn't needed and is redundant. I may be wrong though.

> BTW, is this really Node (effectively Union[Item, Collector]) or just Item?

I checked -- from what I can see, it is an `Item` specifically, not a `Node`. (tracing up, it comes from `pytest_collection_modifyitems` which only takes `Item` from my current understanding).

Reply:
Ahh indeed, thanks!